MES-6607 Team journal candidate results

Cullima opened the pull request:
## Description
[MES-6607](https://jira.dvsacloud.uk/browse/MES-6607)
- Add candidate search results to Test Centre Journals page

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional)
![image](https://user-images.githubusercontent.com/33055124/112664510-f216ae80-8e51-11eb-88bd-5c4770209cf8.png)

Commit:
team journal candidate results

Commit:
clear results on refresh

matthew2564 commented on src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.html:
```diff
@@ -1,15 +1,73 @@
-  <ion-button mode="ios" class="mes-primary-button full-width">
-    <h3>Show booking</h3>
+  <ion-button mode="ios" class="mes-primary-button full-width" [disabled]="!enableShowBookingButton">
+    <h3 (click)="showResults()">Show booking</h3>
```
Need to move click event to ion-button rather than text

matthew2564 commented on src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.ts:
```diff
@@ -12,14 +18,34 @@ type CandidateData = {
   styleUrls: ['candidate-search-card.scss'],
 })
-export class CandidateSearchCardComponent {
+export class CandidateSearchCardComponent implements OnChanges {
```
need to be using the DateTime class throughout this component instead of JS Date

matthew2564 commented on src/components/common/typeahead-dropdown/typeahead-dropdown.ts:
```diff
@@ -18,6 +20,9 @@ export class TypeaheadDropdownComponent {
   property: string;
 
+  @Output()
+  candidateChange = new EventEmitter<string>();
```
think the use of  '< string >' here is incorrect, would assume we are emitting some sort of object i.e. 
{
  name: string,
  slot: TestSlot,
}

However, as this is a common component should probably leave this in its entirety and just use: -

new EventEmitter()

matthew2564 commented on src/app/pages/test-centre-journal/components/candidate-search-card/candidate-search-card.scss:
```diff
@@ -9,3 +9,28 @@
+  background-color: var(--ion-background-color);
+}
+
+.no-results {
```
can be removed due to there being the same class in global.scss

Commit:
Remove type declaration from event emitter in typeahead-dropdown.ts;
Update HTML to move the click event to the ion-button rather than the H3 text; Update SCSS to align with view-journals; Amend TS to use DateTime class;

andrewsetterfield committed:
added class for examiner name

matthew2564 commented on src/app/pages/test-centre-journal/test-centre-journal.module.ts:
```diff
@@ -17,6 +18,8 @@ import { SlotProvider } from '../../providers/slot/slot';
     TestCentreJournalComponentsModule,
     ComponentsModule,
+    TestSlotComponentsModule,
+    TestCentreJournalComponentsModule,
```
Duplicate reference TestCentreJournalComponentsModule

andrewsetterfield committed:
Merge branch 'mes-6607-candidate-results-squashed' of https://github.com/dvsa/des-mobile-app into mes-6607-candidate-results-squashed
